bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -450,9 +778,49 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
          * Returns an array of layers added to the service for example via #addLayer()
          * @return {Mixed[]/Oskari.mapframework.domain.WmsLayer[]/Oskari.mapframework.domain.WfsLayer[]/Oskari.mapframework.domain.VectorLayer[]/Object[]}
          */
-        getAllLayers: function () {
+        getAllLayers: function() {
             return this._loadedLayersList;
         },
+
+        /**
+         * @method getAllLayerGroups
+         * Returns an array of layer groups added to the service
+         * @param {String|Integer} id if defined return only wanted group
+         * @return {Mixed[]/Oskari.mapframework.domain.WmsLayer[]/Oskari.mapframework.domain.WfsLayer[]/Oskari.mapframework.domain.VectorLayer[]/Object[]}
+         */
+        getAllLayerGroups: function(id) {
+            if (id) {
+                var filterFunction = function(group) {
+                    return group.id == id;
+                };
+                var group = this._layerGroups.filter(filterFunction)[0];
+                // group not found
+                // try to get subgroup
+                if (!group) {
+                    for (var i = 0; i < this._layerGroups.length; i++) {
```

Comment:
Is there a reason why you have for (var i = 0;...) and forEach-loops taking turns? I think you should use forEach() if there's not some good reason for not using it.